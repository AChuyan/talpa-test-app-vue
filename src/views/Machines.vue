<template>
  <div class="machines-list-page">
    <h3>List of available machines</h3>
    <machine-list />
  </div>
</template>

<script>
  import MachineList from '../components/mashines/MachineList'

  export default {
    components: {
      MachineList,
    },
    name: 'machines',
  }
</script>
