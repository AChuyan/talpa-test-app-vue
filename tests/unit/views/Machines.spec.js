import { shallowMount } from '@vue/test-utils'
import Machines from '@/views/Machines.vue'

describe('Machine view', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Machines, {
      propsData: {},
      mocks: {},
      stubs: {},
      methods: {},
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  });
});
