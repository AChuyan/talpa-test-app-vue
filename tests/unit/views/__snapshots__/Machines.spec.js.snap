// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MachineList view should renders correctly 1`] = `
<div class="machines-list-page">
  <h3>List of available machines</h3>
  <machine-list-stub></machine-list-stub>
</div>
`;
